src/bridge/MatrixHandler.ts

```diff
@@ -905,14 +905,11 @@ export class MatrixHandler {
             this.eventCache.delete(delKey);
         }
 
-        // Check for the existance of the getSplitMessages method.
-        if (!(ircClient.unsafeClient && ircClient.unsafeClient.getSplitMessages)) {
-            await this.ircBridge.sendIrcAction(ircRoom, ircClient, ircAction);
-            return;
```

Comment:
Instead of waiting for a connection, we formerly send an action and returned here.
Is this change expected?

(The also makes me think if this code is called a ton during a netsplit and queuing many thousands of actions.)

Reply:
Yeah, so I believe this in itself was a bug because getSplitMessages would always be defined on unsafeClient if unsafeClient was defined. I think what likely happens here is that it fails the request because the client isn't ready yet.

That being said, I made this change because of a test failure. That test wasn't failing before.

Reply:
Oh hang on I see `!(ircClient.unsafeClient && ircClient.unsafeClient.getSplitMessages)` would evaluate to true if `unsafeClient` is undefined. Which presumably is not what should be happening here. That'd explain why the test used to pass.

Reply:
>(The also makes me think if this code is called a ton during a netsplit and queuing many thousands of actions.)

Shouldn't be, it's M->I direction so it's only called on new events

Reply:
Isn't a netsplit causing many many leaves and joins on the Matrix side (caused by the bridge)?

Reply:
No, netsplits are many leaves and joins on the irc side. They don't reflect back across the bridge though, so the M->I path is not affected.